src/serial.rs

```diff
@@ -0,0 +1,474 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Portions Copyright 2020 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Provides emulation for Linux serial console.
+//!
+//! This is done by emulating an UART 8250 serial port.
+
+use std::collections::VecDeque;
+use std::io::{Result, Write};
+
+use vmm_sys_util::eventfd::EventFd;
+
+// Register offsets.
```

Comment:
I would name this using the _OFFSET suffix so that you do not have to do the mental map between these names and the fact they're offsets.